src/components/organisms/Settings/Project/Plugin/hooks.ts

```diff
@@ -1,28 +1,38 @@
+import { useMemo } from "react";
+
 import { useLocalState } from "@reearth/state";
+import { useInstallablePluginsQuery, useInstalledPluginsQuery } from "@reearth/gql";
+import { PluginItem } from "@reearth/components/molecules/Settings/Workspace/Plugin/PluginSection";
 
-export default () => {
+export default (projectId: string) => {
   const [{ currentTeam, currentProject }] = useLocalState(s => ({
     currentTeam: s.currentTeam,
     currentProject: s.currentProject,
   }));
 
-  //DELETE_ME: When plugin API is ready
-  const samplePlugins = [
-    {
-      id: "hogehoge",
-      thumbnail: "https://static.dev.reearth.io/assets/01ep431qsvnjndxhan3gwqd1rj.png",
-      title: "Storytelling",
-      isInstalled: true,
-      bodyMarkdown: "# Hoge\n## Fuag",
-    },
-    {
-      id: "fugafuga",
-      thumbnail: "https://static.dev.reearth.io/assets/01ep431qsvnjndxhan3gwqd1rj.png",
-      title: "Storytelling",
-      isInstalled: false,
-      bodyMarkdown: "# Hoge\n## Fuag",
-    },
-  ];
+  const { data: rawPluginsData, loading: pluginLoading } = useInstallablePluginsQuery();
+
+  const { data: rawSceneData, loading: sceneLoading } = useInstalledPluginsQuery({
+    variables: { projectId: projectId ?? "" },
+    skip: !projectId,
+  });
+
+  const installedPluginIds = useMemo(() => {
+    return rawSceneData?.scene?.plugins.map(p => p.plugin?.id);
+  }, [rawSceneData?.scene?.plugins]);
+
+  const plugins = useMemo((): PluginItem[] => {
+    return rawPluginsData
+      ? rawPluginsData?.installablePlugins.map<PluginItem>(p => ({
+          title: p.name,
+          bodyMarkdown: p.description,
+          author: p.author,
+          thumbnailUrl: p.thumbnailUrl,
+          isInstalled: !!installedPluginIds?.includes(p.name), //TODO: After back-end decide how to generate plugin's id, fix here.
+        }))
+      : [];
+  }, [installedPluginIds, rawPluginsData]);
 
-  return { currentTeam, currentProject, plugins: samplePlugins };
+  const loading = sceneLoading || pluginLoading;
```

Comment:
I'm not sure, but I want to ask.
Does this line works correctly?
Loading is a constant and assigned once, what's the purpose of it?

Reply:
It's a const value, but the component will be rendered when the loading state returned by Apollo Query has been changed. So the `loading` will be also updated right? @rot1024

Reply:
Yes, no problem. The component function will be called every render time, so all lines are also executed every time. That is the same as normal functions of JavaScript. Only the value returned by the hooks may change each time the component is rendered.